pkg/addon/versioned_registry.go

```diff
@@ -146,7 +147,7 @@ func (i versionedRegistry) loadAddon(ctx context.Context, name, version string)
 	sort.Sort(sort.Reverse(versions))
 	addonVersion, availableVersions := chooseVersion(version, versions)
 	if addonVersion == nil {
-		return nil, fmt.Errorf("specified version %s not exist", version)
```

Comment:
the version info is useful, do we really need to remove it ?

Reply:
Version specified by user is already known by user themselves. Using "specified version" to refer to the argument passed by use is enough. And this will solve the security problem.